assets/dev/js/editor/document/elements/commands/paste.js

```diff
@@ -20,106 +20,115 @@ export class Paste extends CommandHistory {
 
 	apply( args ) {
 		const { at, rebuild = false, storageKey = 'clipboard', containers = [ args.container ], options = {} } = args,
-			storageData = elementorCommon.storage.get( storageKey ),
-			result = [];
+			storageData = elementorCommon.storage.get( storageKey );
+
+		let result = [];
 
 		// Paste on "Add Section" area.
 		if ( rebuild ) {
-			// Paste at each target.
-			containers.forEach( ( targetContainer ) => {
-				let index = 'undefined' === typeof at ? targetContainer.view.collection.length : at;
-
-				storageData.forEach( ( model ) => {
-					switch ( model.elType ) {
-						case 'section': {
-							// If is inner create section for `inner-section`.
-							if ( model.isInner ) {
-								const section = $e.run( 'document/elements/create', {
-									container: targetContainer,
-									model: {
-										elType: 'section',
-									},
-									columns: 1,
-									options: {
-										at: index,
-										edit: false,
-									},
-								} );
-
-								// `targetContainer` = first column at `section`.
-								targetContainer = section.view.children.findByIndex( 0 ).getContainer();
-							}
-
-							// Will be not affected by hook since it always have `model.elements`.
-							result.push( this.pasteTo(
-								[ targetContainer ],
-								[ model ],
-								{
-									at: index,
-									edit: false,
-								} )
-							);
-							index++;
-						}
-						break;
+			result = this.rebuild( containers, storageData, at );
+		} else {
+			if ( undefined !== at ) {
+				options.at = at;
+			}
+
+			result.push( this.pasteTo( containers, storageData, options ) );
+		}
 
-						case 'column': {
-							// Next code changed from original since `_checkIsEmpty()` was removed.
+		if ( 1 === result.length ) {
+			return result[ 0 ];
+		}
+
+		return result;
+	}
+
+	rebuild( containers, data, at ) {
+		// Paste at each target.
+		const result = [];
+
+		containers.forEach( ( targetContainer ) => {
+			let index = 'undefined' === typeof at ? targetContainer.view.collection.length : at;
+
+			data.forEach( ( model ) => {
+				switch ( model.elType ) {
```

Comment:
Maybe extract each case into a method with an explicit name?

Reply:
Out of scope, i don't want to modify the source, just make to make it extendable.